jlm/rvsdg/region.cpp

```diff
@@ -515,6 +515,39 @@ Region::ToString(const util::Annotation & annotation, char labelValueSeparator)
   return util::strfmt(annotation.Label(), labelValueSeparator, value);
 }
 
+static size_t
+computeDepth(const Node & node, std::unordered_map<const Node *, size_t> & depthMap)
+{
+  if (depthMap.find(&node) != depthMap.end())
+  {
+    return depthMap[&node];
+  }
+
+  size_t depth = 0;
+  for (auto & input : node.Inputs())
```

Comment:
I was about to say to use std::max_element on iterator range instead -- but it doesn't apply here as this doesn't deal with cached values